kanidmd/src/lib/interval.rs

```diff
@@ -24,4 +29,51 @@ impl IntervalActor {
             }
         });
     }
+
+    pub fn start_online_backup(server: &'static QueryServerReadV1, cfg: &OnlineBackup) {
+        let outpath = cfg.path.to_owned();
+        let schedule = cfg.schedule.to_owned();
+        let versions = cfg.versions;
+
+        // TODO: add some checks arount the provided cron pattern .any() etc.
+        let cron_expr = match schedule.as_str().parse::<CronExpr>() {
+            Ok(ce) => ce,
+            Err(e) => {
+                info!(
+                    "Online backup schedule parse error {}. Pattern set to < 00 22 * * * >",
+                    e
+                );
+                "00 22 * * *".parse::<CronExpr>().unwrap()
+            }
+        };
+
+        info!(
+            "Online backup schedule parsed as: {}",
+            cron_expr.describe(English::default())
+        );
+
+        tokio::spawn(async move {
+            let ct = Utc::now();
+            let cron = Cron::new(cron_expr.clone());
+
+            let cron_iter = cron.clone().iter_after(ct);
+            for next_time in cron_iter {
+                // +1 to have even times, but we might anyway cut away the seconds from the timestamp.
+                let wait_seconds = 1 + (next_time - Utc::now()).num_seconds() as u64;
+                info!(
+                    "Online backup next run on {}, wait_time = {}s",
+                    next_time, wait_seconds
+                );
+
+                sleep(Duration::from_secs(wait_seconds)).await;
```

Comment:
This basicly ends up doing the same as sleep_until I think anyway, so this is okay.